src/main/kotlin/app/revanced/patches/youtube/layout/translations/TranslationsPatch.kt

```diff
@@ -1,41 +1,97 @@
 package app.revanced.patches.youtube.layout.translations
 
 import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchException
+import app.revanced.patcher.patch.options.PatchOption.PatchExtensions.stringPatchOption
+import app.revanced.patches.shared.translations.APP_LANGUAGES
 import app.revanced.patches.shared.translations.TranslationsUtils.copyXml
+import app.revanced.patches.shared.translations.TranslationsUtils.updateStringsXml
 import app.revanced.patches.youtube.utils.compatibility.Constants.COMPATIBLE_PACKAGE
 import app.revanced.patches.youtube.utils.settings.SettingsPatch
 import app.revanced.util.patch.BaseResourcePatch
+import java.io.File
 
-@Suppress("unused")
+// Array of supported RVX languages, each represented by its language code.
+val LANGUAGES = arrayOf(
+    "ar",
+    "el-rGR",
+    "es-rES",
+    "fr-rFR",
+    "hu-rHU",
+    "it-rIT",
+    "ja-rJP",
+    "ko-rKR",
+    "pl-rPL",
+    "pt-rBR",
+    "ru-rRU",
+    "tr-rTR",
+    "uk-rUA",
+    "vi-rVN",
+    "zh-rCN",
+    "zh-rTW"
+)
+
+@Suppress("DEPRECATION", "unused")
 object TranslationsPatch : BaseResourcePatch(
     name = "Translations",
-    description = "Adds Crowdin translations for YouTube.",
+    description = "Add Crowdin translations for YouTube.",
     dependencies = setOf(SettingsPatch::class),
     compatiblePackages = COMPATIBLE_PACKAGE
 ) {
+    private var CustomLanguage by stringPatchOption(
+        key = "CustomLanguage",
+        default = "",
+        title = "Custom language file",
+        description = """
+            The file path to the strings.xml file.
```

Comment:
```suggestion
            The path to the strings.xml file.
```

Also, maybe `strings.xml` should have some sort of quotes around it. Such as like this:

```
The path to the 'strings.xml' file.
```